File: KHost.App/ClientApp/src/app/models/SingerPerformance.ts
import { ModelWithId } from "./ModelWIthId";
import { Singer } from "./Singer";
import { Song } from "./Song";
import { Venue } from "./Venue";

export class SingerPerformance implements ModelWithId {

    id?: number;

    singerId?: number;

    singer?: Singer;
    
    songId?: number;

    song?: Song;
    
    venueId?: number;

    venue?: Venue;
    
    date?: Date;
}

File: KHost.Common/Models/SingerPerformance.cs
using System;

namespace KHost.Common.Models
{
    public class SingerPerformance : BaseModel, IModelWithId
    {
        public int? Id { get; set; }

        public int SingerId { get; set; }

        public int SongId { get; set; }

        public int VenueId { get; set; }

        public DateTime Date { get; set; }

        public override SingerPerformance Clone() => (base.Clone() as SingerPerformance)!;
    }
}
